components/admin_console/team_channel_settings/channel/details/channel_details.jsx

```diff
@@ -132,35 +134,68 @@ export default class ChannelDetails extends React.Component {
     hideRemoveUsersModal = () => {
         this.setState({showRemoveConfirmation: false});
     }
-    showRemoveUsersModal = () => {
+
+    hideConvertChannelModal = () => {
+        this.setState({showConvertConfirmation: false});
+    }
+
+    onSave = () => {
         if (this.state.usersToRemove > 0) {
```

Comment:
Ok, this approach have a flaw, if you have changes in both sides, you only be requested for confirmation once, for the Remove confirmation, then, the modal is going to run directly `handleSubmit` without the confirmation about channel privacy.

Reply:
Yes my assumption was that if the sync toggle is enabled the user knows the channel is going to be set to private also because the privacy setting toggle disappears.
What we could do I guess would be to show a third modal warning of both changes.

Reply:
Yes, probably a combined one would be the best option from the UX perspective.